pkg/uptimerobot.go

```diff
@@ -215,11 +223,34 @@ func (c *Client) GetMonitorByID(ID int64) (Monitor, error) {
 
 // GetMonitors returns a slice of Monitors representing the existing monitors.
 func (c *Client) GetMonitors() (monitors []Monitor, err error) {
-	r := Response{}
-	if err := c.MakeAPICall("getMonitors", &r, Params{}); err != nil {
-		return monitors, err
+	offset := 0
+	limit := 50
+
+	for {
+		r := Response{}
+		params := Params{
+			"offset": strconv.Itoa(offset),
+			"limit":  strconv.Itoa(limit),
+		}
+		if err := c.MakeAPICall("getMonitors", &r, params); err != nil {
+			break
+		}
+
+		monitors = append(monitors, r.Monitors...)
+
+		if r.Error != nil {
```

Comment:
Should this check come before adding the returned value to the results? The conventional semantics in Go is that if the error is non-nil, then the data value is worthless and shouldn't be used.